aws-iot-billinggroup/src/test/java/software/amazon/iot/billinggroup/CreateHandlerTest.java

```diff
@@ -0,0 +1,199 @@
+import software.amazon.awssdk.services.iot.model.DescribeBillingGroupRequest;
+import software.amazon.awssdk.services.iot.model.DescribeBillingGroupResponse;
+import software.amazon.awssdk.services.iot.model.InternalException;
+import software.amazon.awssdk.services.iot.model.InvalidRequestException;
+import software.amazon.awssdk.services.iot.model.ResourceAlreadyExistsException;
+import software.amazon.awssdk.services.iot.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.iot.model.ThrottlingException;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class CreateHandlerTest extends AbstractTestBase {
+    final CreateHandler handler = new CreateHandler();
+
+    @Test
+    public void handleRequest_Success() {
+        final ResourceModel model = ResourceModel.builder()
+                .billingGroupName(BG_Name)
+                .id(BG_ID)
+                .billingGroupProperties(software.amazon.iot.billinggroup.BillingGroupProperties.builder()
+                        .billingGroupDescription(BILLING_GROUP_DESCRIPTION)
+                        .build())
+                .build();
+        final ResourceHandlerRequest<ResourceModel> request = defaultRequestBuilder(model).build();
+
+        when(iotClient.describeBillingGroup(any(DescribeBillingGroupRequest.class)))
+                .thenThrow(ResourceNotFoundException.class);
+        when(iotClient.createBillingGroup(any(CreateBillingGroupRequest.class)))
+                .thenReturn(CreateBillingGroupResponse.builder()
+                        .billingGroupArn(BG_ARN)
+                        .billingGroupId(BG_ID)
+                        .billingGroupName(BG_Name)
+                        .build());
+
+        final ProgressEvent<ResourceModel, CallbackContext> response =
+                handler.handleRequest(proxy, request, new CallbackContext(), proxyClient, LOGGER);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+        assertThat(response.getCallbackContext()).isNull();
+        assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+        assertThat(response.getResourceModel()).isEqualTo(request.getDesiredResourceState());
+        org.assertj.core.api.Assertions.assertThat(response.getResourceModels()).isNull();
+        assertThat(response.getMessage()).isNull();
+        assertThat(response.getErrorCode()).isNull();
+        assertThat(response.getResourceModel().getId()).isEqualTo(BG_ID);
+    }
+
+
+    @Test
+    public void handleRequest_Success_No_Name() {
```

Comment:
does this mean that the handler will generate a BillingGroup name if not provided by customer?

Reply:
Yes, if billing group name is not provided in the template. It will be generated.